src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -1704,6 +1704,9 @@ private bool IsRef(BoundExpression receiver)
                 case BoundKind.Call:
                     return ((BoundCall)receiver).Method.RefKind != RefKind.None;
 
+                case BoundKind.FunctionPointerInvocation:
+                    return ((BoundFunctionPointerInvocation)receiver).FunctionPointer.Signature.RefKind != RefKind.None;
```

Comment:
This change is tested by `CodeGenFunctionPointersTests.FunctionPointerReturnTypeConstrainedCallVirtIfNotRef`. It's used by `EmitCall` when determining whether the receiver needs to be a constrained or boxed for a call virt.